src/objects/tests/v1/test_auth_fields.py

```diff
@@ -321,6 +347,37 @@ def test_retrieve_query_fields_not_allowed(self):
             ["'fields' query parameter has invalid or unauthorized values: 'uuid'"],
         )
 
+    def test_list_no_allowed_fields(self):
+        PermissionFactory.create(
+            object_type=self.object_type,
+            mode=PermissionModes.read_only,
+            token_auth=self.token_auth,
+            use_fields=True,
+            fields={"2": ["url", "type", "record"]},
+        )
+        PermissionFactory.create(
+            object_type=self.other_object_type,
+            mode=PermissionModes.read_only,
+            token_auth=self.token_auth,
+            use_fields=True,
+            fields={"2": ["url", "uuid", "record"]},
+        )
+        ObjectRecordFactory.create(
+            object__object_type=self.object_type, data={"name": "some"}, version=1
+        )
+        ObjectRecordFactory.create(
+            object__object_type=self.other_object_type,
+            data={"name": "other"},
+            version=1,
+        )
+        url = reverse("object-list")
+
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.json(), [{}, {}])
+        self.assertIn("x-unauthorized-fields", response._headers)
```

Comment:
I don't see this header documented in the API spec

Reply:
Thanks for catching it. `x-unauthorized-fields` header is added to the OAS